service/routes.py

```diff
@@ -46,6 +51,51 @@ def index():
     url = request.base_url + 'shopcarts'  # url_for('shopcarts')
     return jsonify(name='Shopcart Demo REST API Service', version='1.0', url=url, data=data), status.HTTP_200_OK
 
+######################################################################
+# ADD A NEW SHOPCART
+######################################################################
+@app.route("/shopcarts", methods=["POST"])
+def create_shopcarts():
+    """
+    Creates a Shopcart
+    This endpoint will create a Shopcart based the data in the body that is posted
+    """
+    app.logger.info("Request to create a shopcart")
+    check_content_type("application/json")
+    shopcart = Shopcart()
+    shopcart.deserialize(request.get_json())
+    shopcart.create()
+    message = shopcart.serialize()
+    #TODO: location_url = url_for("get_shopcarts", shopcart_id=shopcart.id, _external=True)
+    location_url = request.base_url + 'shopcarts'
+
+    app.logger.info("Shopcart with ID [%s] created.", shopcart.id)
+    return make_response(
+        jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
+    )
+
+######################################################################
+# ADD A NEW SHOPCARTITEM
+######################################################################
+@app.route("/shopcartitems", methods=["POST"])
+def create_shopcart_items():
+    """
+    Creates a ShopcartItem
+    This endpoint will create a ShopcartItem based the data in the body that is posted
+    """
+    app.logger.info("Request to create a shopcart item")
+    check_content_type("application/json")
+    shopcart_item = ShopcartItem()
+    shopcart_item.deserialize(request.get_json())
```

Comment:
Do we need to check if `sid` exists?

Reply:
wouldn't the deserialize function throw a DataValidationError exception though? I can have it raise an error as well if sid is None

Reply:
Sorry, I mean the shopcart that the `sid` refers to. Do we need to check if that shopcart exists?

Reply:
well can't do that until the get functionality is added, could add as another issue

Reply:
Okay, thanks!